apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Analyzers/GoogleTraceBeforeMvcAnalyzer.cs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Google.Cloud.Diagnostics.AspNetCore.Analyzers
+{
+    // TODO: Support for VB as well?
+    // TODO: Add a code fix provider?
+
+    /// <summary>
+    /// Analyzer which checks to make sure the earliest usage of the UseGoogleTrace extension method on
+    /// an IApplicationBuilder symbol happens before the earliest usage any UseMvc... extension method
+    /// on the same symbol. Chains of method calls all on an IApplicationBuilder are assumed to all be
+    /// called on the same instance.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class GoogleTraceBeforeMvcAnalyzer : DiagnosticAnalyzer
+    {
+        public const string RuleId = "GCP0005";
+
+        private const string IApplicationBuilderFullName = "Microsoft.AspNetCore.Builder.IApplicationBuilder";
+        private const string UseGoogleTrace = "UseGoogleTrace";
+        private const string UseMvc = "UseMvc";
+
+        private static readonly LocalizableString Title =
```

Comment:
I *think* we should capitalize Trace here and below

Reply:
I think you're right. Done